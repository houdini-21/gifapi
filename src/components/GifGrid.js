import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);
  return (
    <>
      {loading && <h1>load</h1>}
      <div className="box__items">
        {images.map((item) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GifGrid;
