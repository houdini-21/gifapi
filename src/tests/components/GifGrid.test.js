import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en el componente gif grid', () => {
  test('hacer snapshot del componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    let wrapper = shallow(<GifGrid category="silicon valley" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando carga imagenes useFetchGifs', () => {
    const imgs = [
      {
        id: 'abc',
        url: 'houdini.com',
        title: 'lorem ipsum',
      },

      {
        id: 'c',
        url: 'houdini.com',
        title: 'lorem ipsum',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false,
    });

    let wrapper = shallow(<GifGrid category="silicon valley" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
  });
});
