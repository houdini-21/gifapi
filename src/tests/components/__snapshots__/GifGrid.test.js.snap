// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente gif grid debe de mostrar items cuando carga imagenes useFetchGifs 1`] = `
<Fragment>
  <div
    className="box__items"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="lorem ipsum"
      url="houdini.com"
    />
    <GifGridItem
      id="c"
      key="c"
      title="lorem ipsum"
      url="houdini.com"
    />
  </div>
</Fragment>
`;

exports[`pruebas en el componente gif grid hacer snapshot del componente 1`] = `
<Fragment>
  <h1>
    load
  </h1>
  <div
    className="box__items"
  />
</Fragment>
`;
